OpenRA.Mods.Common/Scripting/Properties/DemolitionProperties.cs

```diff
@@ -19,21 +19,22 @@ namespace OpenRA.Mods.Common.Scripting
 	[ScriptPropertyGroup("Combat")]
 	public class DemolitionProperties : ScriptActorProperties, Requires<IMoveInfo>, Requires<DemolitionInfo>
 	{
-		readonly DemolitionInfo info;
+		readonly Demolition[] demolitions;
 
 		public DemolitionProperties(ScriptContext context, Actor self)
 			: base(context, self)
 		{
-			info = Self.Info.TraitInfo<DemolitionInfo>();
+			demolitions = Self.TraitsImplementing<Demolition>().ToArray();
 		}
 
 		[ScriptActorPropertyActivity]
 		[Desc("Demolish the target actor.")]
 		public void Demolish(Actor target)
 		{
 			// NB: Scripted actions get no visible targetlines.
-			Self.QueueActivity(new Demolish(Self, Target.FromActor(target), info.EnterBehaviour, info.DetonationDelay,
-				info.Flashes, info.FlashesDelay, info.FlashInterval, info.DamageTypes, null));
+			var demolition = demolitions.FirstEnabledTraitOrDefault();
+			if (demolition != null)
+				Self.QueueActivity(demolition.GetDemolishActivity(Self, Target.FromActor(target)));
```

Comment:
This breaks the behaviour mentioned in the comment above.

Reply:
hmm, I though not calling ShowTargetLines was enough. Fixed